cartographer/pose_graph/node/imu_calibration.cc

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cartographer/pose_graph/node/imu_calibration.h"
+
+namespace cartographer {
+namespace pose_graph {
+
+ImuCalibration::ImuCalibration(const NodeId& node_id, bool constant,
+                               double gravity_constant,
+                               const Eigen::Quaterniond& orientation)
+    : Node(node_id, constant),
+      gravity_constant_(gravity_constant),
+      orientation_{{orientation.w(), orientation.x(), orientation.y(),
```

Comment:
I assume that the order is important but it comes at a surprise. Could you add a comment somewhere why that is? Alternatively, could we keep the Eigen::Quaterniond format?

Also the same below in Pose3D. Maybe we can have a converter function?

Reply:
I just kept Cartographer's format. But I agree, let's keep Eigen's.